substrate/pkg/apis/v1alpha1/substrate.go

```diff
@@ -50,3 +57,19 @@ type SubnetSpec struct {
 	CIDR   string
 	Public bool
 }
+
+func (s *Substrate) IsReady() bool {
```

Comment:
You should use a living condition set

Reply:
I used in the next revision of the PR but I am not sure whats the benefit and even if I am using it the right way.

Reply:
I think this isn't the right way, but I'd have to play with it and/or dig into tekton.